src/cmd/troubleshoot/checks_executor.go

```diff
@@ -0,0 +1,48 @@
+package troubleshoot
+
+import (
+	tserrors "github.com/Dynatrace/dynatrace-operator/src/cmd/troubleshoot/errors"
+)
+
+type Result int
+
+const (
+	PASSED = iota + 1
+	FAILED
+	SKIPPED
+)
+
+var results = map[string]Result{}
+
+func runChecks(troubleshootCtx *troubleshootContext, checks []Check) error {
```

Comment:
This function should also take care of the execution order so that all pre-requisites are executed before their dependences.

Reply:
that would be overkill logic vs maintaining a reasonable list of `Checks` that are in the correct order.

having the checks in the correct order in code would be mandatory IMO to keep the code readable anyway.

Reply:
Good catch, Stefan, It's directed acyclic graph, so traveling the graph can be done smarter.
I also think it's an overkill for the moment. We can improve it later.

Reply:
Do it right or not at all. The way how it's currently implemented it might happen that checks are not executed if they are configured in the wrong order, which is a potential pitfall for future devs that come new to the project. IMHO that's a bug which needs to be resolved.

Reply:
It is done right at the moment, since building a graph and resolving it would be overkill and therefore wrong since a solution that is over-complicated for the intended purpose is wrong by definition. We can think about that when we have, like, 50 checks, but with the amount of checks we have now a list is more maintainable